From 2b3e35d98bcaad9218ec3d4ab91e93022f76b6dd Mon Sep 17 00:00:00 2001
From: Johannes Berg <johannes.berg@intel.com>
Date: Thu, 11 Jan 2024 18:17:39 +0200
Subject: [PATCH] wifi: mac80211_hwsim: advertise 15 simultaneous links
Git-commit: 2b3e35d98bcaad9218ec3d4ab91e93022f76b6dd
Patch-mainline: v6.9-rc1
References: bsc#1230505

Advertise MLD capabilities and operations in AP mode that
say that up to 15 links are supported simultaneously.

Signed-off-by: Johannes Berg <johannes.berg@intel.com>
Reviewed-by: Gregory Greenman <gregory.greenman@intel.com>
Reviewed-by: Ilan Peer <ilan.peer@intel.com>
Signed-off-by: Miri Korenblit <miriam.rachel.korenblit@intel.com>
Link: https://msgid.link/20240111181514.52a1d48b67e6.Ie459df742944d24d6401683d54d2f3ac44834803@changeid
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/net/wireless/virtual/mac80211_hwsim.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/virtual/mac80211_hwsim.c b/drivers/net/wireless/virtual/mac80211_hwsim.c
index 304dc96fc3bb..403d892c0468 100644
--- a/drivers/net/wireless/virtual/mac80211_hwsim.c
+++ b/drivers/net/wireless/virtual/mac80211_hwsim.c
@@ -4993,9 +4993,11 @@ static const u8 iftypes_ext_capa_ap[] = {
 	 [9] = WLAN_EXT_CAPA10_TWT_RESPONDER_SUPPORT,
 };
 
-#define MAC80211_HWSIM_MLD_CAPA_OPS FIELD_PREP_CONST( \
-	IEEE80211_MLD_CAP_OP_TID_TO_LINK_MAP_NEG_SUPP, \
-	IEEE80211_MLD_CAP_OP_TID_TO_LINK_MAP_NEG_SUPP_SAME)
+#define MAC80211_HWSIM_MLD_CAPA_OPS				\
+	FIELD_PREP_CONST(IEEE80211_MLD_CAP_OP_TID_TO_LINK_MAP_NEG_SUPP, \
+			 IEEE80211_MLD_CAP_OP_TID_TO_LINK_MAP_NEG_SUPP_SAME) | \
+	FIELD_PREP_CONST(IEEE80211_MLD_CAP_OP_MAX_SIMUL_LINKS, \
+			 IEEE80211_MLD_MAX_NUM_LINKS - 1)
 
 static const struct wiphy_iftype_ext_capab mac80211_hwsim_iftypes_ext_capa[] = {
 	{
-- 
2.43.0

